include/exadg/compressible_navier_stokes/spatial_discretization/kernels_and_operators.h

```diff
@@ -1012,9 +1012,10 @@ class ViscousOperator
     scalar
     get_penalty_parameter(FaceIntegratorScalar & fe_eval_m, FaceIntegratorScalar & fe_eval_p) const
   {
-    scalar tau = std::max(fe_eval_m.read_cell_data(array_penalty_parameter),
-                          fe_eval_p.read_cell_data(array_penalty_parameter)) *
-                 IP::get_penalty_factor<Number>(degree, data.IP_factor) * nu;
+    scalar tau =
+      std::max(fe_eval_m.read_cell_data(array_penalty_parameter),
+               fe_eval_p.read_cell_data(array_penalty_parameter)) *
+      IP::get_penalty_factor<dim, Number>(degree, fe_eval_m.get_matrix_free(), data.IP_factor) * nu;
```

Comment:
I must admit that I don't really like that `get_penalty_factor` gets a `MatrixFree` object but I don't have a better solution.

Reply:
I have also implemented the mentioned alternative solution. But I decided for this one, since the `calculate_penalty_parameter` function also gets a `MatrixFree` function: https://github.com/exadg/exadg/blob/7edf07c5a30582accdd72ce2af833caa51382107/include/exadg/operators/interior_penalty_parameter.h#L41-L44

But I am okay with both solutions.

Reply:
I agree with Peter on this point. With the `calculate_penalty_parameter` function it is different because this function needs different kinds of information from the `MatrixFree` object. As long as the only information necessary for `get_penalty_factor` is if simplex or hypercube, we should determine this at the calling side and only pass this information.

Reply:
Okay, I will change it accordingly.